### ISSUE TITLE
Warning: componentWillMount has been renamed, and is not recommended for use.

### ISSUE THREAD

**Issue by PierrickLozach** [styled-components/styled-components#5532]
Even though I used the latest version (@6.1.14), I still get this warning: `Warning: componentWillMount has been renamed, and is not recommended for use.` when using something like `styled.div`:

```
import styled from 'styled-components';

const StylesWrapper = styled.div`
  .droppable-wrapper-no-dropping {
    position: relative;
    opacity: 1;
    border: none;
    border: 1px solid transparent;
  }

  .droppable-wrapper-dropping {
    position: relative;
    opacity: 0.5;
    border: 1px dashed ${(props) => props.theme.primary};
    border-radius: 0.5rem;
    overflow: hidden;
  }
`;
```

I use React@18.2.0.

I looked at existing issues but they seem to have been fixed a few years ago. Could you help me get rid of this warning?

Thanks!

**Comment by quantizor** [comment]
Closing — this is a support question, stale with no reproduction, or not actionable as a bug report. For help, please use GitHub Discussions or Stack Overflow.
